### ISSUE TITLE
Is there a keyboard shortcut to move cursor focus from cell to cell?

### ISSUE THREAD

**Issue by richrad** [HappenApps/Quiver#1]
If not, I'd love one!

**Comment by bradley-curran** [comment]
Can't wait! :)

**Comment by ylian** [comment]
Already having a working build. Only have a couple edge cases that aren't working properly. I will release this in a few days.

**Comment by LeonardoGentile** [comment]
+1

**Comment by ylian** [comment]
Quiver 2.1 is almost ready. Fixing the last few bugs. Lots of improvements.

**Comment by matthewwoo** [comment]
Close to being released haha it's killing me, but I love the product so I shouldn't complain too much.

**Comment by ylian** [comment]
Appreciate your patience. Just a couple more things on my list. Here is the current release notes in case you wonder.

New features:
	•	Keyboard navigation: now you can use the up, down, left and right arrows to move between cells. You can also use the right arrow key to navigate from the note list to the note, and the escape key to come back.
	•	Added syntax highlighting support for 17 languages: Cirru, Dockerfile, Eiffel, Elixir, Elm, Gcode, Gherkin, Gitignore, Io, N-Triples, Praat, SAS, Smarty, SPARQL, SPSS, Turtle, Vala
	•	Added a setting to scroll the note content past end while editing
	•	Added settings for enabling autocompletion and tab triggers in code cells
	•	User settings (code cell theme, tab size, etc.) are now applied to the preview and presentation mode.
	•	Added keyboard shortcuts for inserting lists and headings
	•	Added a yellow highlighter to the text cell toolbar
	•	Added a keyboard shortcut (Cmd+K) to make a URL into a link
	•	Added a keyboard shortcut (Opt-Cmd-/) to change language in a code cell
	•	Support checkboxes in text cells
	•	Added a toolbar button to insert an image in a text cell
	•	Table support in Markdown cells
	•	Support GFM task lists in Markdown cells
	•	Pretty print JSON, XML, CSS and SQL
	•	Set tab sizes for all cell types
	•	Edit CSS for preview, presentation, and exported HTML
	•	Added a setting to show line numbers in code cells
	•	Added settings to show/hide invisibles, indent guides in code cells
	•	Added options to use Vim or Emacs key binding in code cells, Markdown cells and LaTeX cells
	•	Added a setting to use GFM linebreaks in Markdown cells
	•	Paste note links into a Markdown cell
	•	Inline LaTeX now works in Markdown cells
	•	Add application-wide custom macros to LaTeX cells
	•	Added navigation buttons and keyboard shortcuts (left and right arrow keys) to the presentation mode
	•	Export all your settings and CSS customizations into a single JSON file. This makes it easy to sync settings across multiple computers.
	•	Import Markdown and plain text files
	•	Export to Markdown: Markdown cells are concatenated, code cells become fenced code blocks.
	•	When exporting a notebook to HTML, all the pages are now linked together.

Enhancements:
	•	Quiver now remembers the last opened note and reopens it upon launch.
	•	Tags are now case-sensitive.
	•	Removed the alert when moving notes to trash.
	•	Changed the keyboard shortcut for deleting a note to “cmd+delete” to avoid accidental deletions
	•	Added a “Rename Notebook” menu item to the notebook context menu
	•	Added a “Add to Shortcuts” menu item to the note context menu

Bug fixes:
	•	Fixed a bug that sometimes notebook names were not saved.
	•	Fixed a bug that the tutorial notebook couldn’t be permanently deleted.
	•	Fixed an issue that Markdown preview sometimes gets out of sync.
	•	Fixed Markdown preview styles: paragraphs are now properly separated.
	•	Fixed an issue with inserting line breaks in a LaTeX cell.
	•	Fixed a bug that changing the code cell language in a floating window didn’t update the code cell in the main window.

**Comment by michael-jarosik** [comment]
Wow that sounds really comprehensive - can't wait for the update. Just one question -- when you're typing in a cell, will hitting an arrow key (without a modifier key) really move you to a different cell? That sounds like it might be an issue -- what if you want to move back a few characters in the cell or up/down one line? I'm guessing you mean that hitting Cmd + Arrow will move between cells but want to make sure.

**Comment by ylian** [comment]
Hitting the arrow key inside a cell will work as usual. Only when you are at the end of a cell, hitting the right or down arrow moves to the beginning of the next cell. Same for when you are at the beginning of a cell.

**Comment by michael-jarosik** [comment]
Thx for responding. I'd like to hear other people's thoughts on this also, but I still think a modifier key should be used for navigating between cells.

Here's my logic: when you're typing in a cell, you're almost always at the end of the cell (unless you intentionally put in extra spaces or characters). Let's say you're typing away in a code cell and notice a typo in the second to last line -- the one just above where you're typing. I would expect that hitting the Up Arrow key would take me to that line so I can make the edit. If I'm understanding you correctly, hitting the Up Arrow key would instead take me to the previous cell.

I could probably get accustomed to adding a `//` or something at the end of every cell to make it easier to control whether the arrow keys will navigate or move the cursor, but that seems less-ideal than hitting `Cmd + Arrow` to navigate and the arrow keys alone to move the cursor.

Again, happy to hear your thoughts and others' and either way the new version sounds awesome.

**Comment by michael-jarosik** [comment]
Oh okay, had to think about that for a minute but I think it should work. So let's say I want to navigate up three cells from the current cell: what keys would I type to get there?

**Comment by ylian** [comment]
It's simple. Just keep pressing the up arrow key.

To make things clear to understand, here is what I mean:

1. If the cursor is within a cell, pressing up key moves the cursor one line up, until it reaches the first line in the cell. Press the up key again, the cursor goes to the beginning of the cell, like in any text editor.
2. If you press the up key again (the cursor is now at the beginning of the cell), the cursor goes to the end of the cell above.
3. Press the up key again, the cursor goes up one line, just as in Step 1.

Basically the cell abstraction is completely transparent. You move cursor the same way as you would in any text editor.

**Comment by ylian** [comment]
I think I get your point: you want a faster way to move between cells. Cmd+arrow would work as a faster version of just arrow keys. This seems to be a standard shortcut in most text editors, even code editors like Ace. Cmd+Up goes to the beginning, and Cmd+Down goes to the end.

So with cells, Cmd+Up goes to the beginning of the current cell, Cmd+Up again goes to the end of the cell above, Cmd+Up again goes to the start of the cell above.

Similarly, Cmd+Down goes to the end of the current cell, Cmd+Down again goes to the start of the next cell, Cmd+Down again goes to the end of the next cell. This is much faster than moving line by line.

Let me know what you think.

**Comment by michael-jarosik** [comment]
Yes I think the ability to "jump around" between cells would be great for user experience. I do see your point also about using just the arrow keys -- it would feel more like a standard Word document. Hopefully others can weigh in on this as well to form a consensus.

**Comment by ylian** [comment]
I think we should have both: arrow keys and Cmd+Arrow.

Cmd+Arrow is just a faster version of using only arrow keys. Seems quite natural to me.

**Comment by michael-jarosik** [comment]
That sounds fantastic -- was going to suggest it but didn't want to be greedy. :-)

**Comment by hettiger** [comment]
Really great stuff ylian and I'm a big fan of arrow keys + Cmd+Arrow. I'm glad I've spent the money on this fantastic app :+1:

**Comment by ylian** [comment]
All implemented. Hopefully Apple will approve v2.1 on Monday.

**Comment by ylian** [comment]
Implemented in Quiver 2.1. Use up, down, left, right arrows to move between cells. Cmd+arrow keys for faster navigation.
